app/controllers/application_controller.rb

```diff
@@ -4,6 +4,20 @@ class ApplicationController < ActionController::Base
 
   before_action :turbo_frame_request_variant
 
+  helper_method :cart_items_count
+
+  def cart_items_count
+    Cart::CartItemsCountService.new(current_cart).call
+  end
+
+  def current_cart
+    return Order.find(session[:cart_id]) if session[:cart_id]
+
+    cart = Order.create
+    session[:cart_id] = cart.id
+    cart
```

Comment:
May i keep this method in app controller , as for me its very easy to use in controller, services and also in my helper method. Сoncerning  orders in database app just left then in db, for now it looks a vulnerability. To my mind good to implement logic that clean old and unfinished orders , but in future steps because i'm going to have a state column that can help me to check order. As  I suppose it is possible to create scheduled job that search for old and unfinished job and than remove it from db.

Reply:
You can keep this method, but logic from this method move to service object. Also i sad not about removing uncompleted orders by job it is sure thing. i mean what if user has uncompleted order and he close browser and then open again he lost their cart. I think you need to add this logic to cart service. Another point: what if guest add books to order and then log in, order still without user. I think you need to add this logic to login. And for avoid unnecessary order creation pls create order when first book adds to cart